cli/utils/iso_util.py

```diff
@@ -169,38 +170,46 @@ def upload_iso_to_media_repository(config, cookies, iso_dir, iso_file_name, sys_
         uploaded, vios_uuid = is_iso_uploaded(
             config, cookies, iso_file_name, sys_uuid, vios_uuid_list)
         if uploaded:
+            logger.debug(f"Bootstrap ISO already uploaded to VIOS '{vios_uuid}'")
             return vios_uuid
+    if "_pimc" in iso_file_name:
+        exists, _, lpar_uuid = partition.check_partition_exists(config, cookies, sys_uuid)
+        # Re-run scenario: If lpar is already activated but launch flow failed during monitoring or app_check stage in previous run. Skip reupload of cloudinit iso 
+        if exists:
```

Comment:
If lpar fails after activation due to network misconfig, then in the next attempt if we correct it and retrigger launch it will skip reupload rt? only option left is to destroy and launch. Would we still need to skip reupload of cloud-init since the operation is not of much overhead?